src/pages/balances/BalancesPage.tsx

```diff
@@ -120,86 +118,196 @@ export function OtherBalanceChart() {
-        </Card>
-        <Card className="py-4">
-          <OtherBalanceChart />
-        </Card>
-      </div>
-    </div>
-  )
+    try {
+      // Fetch both credit and debit balances concurrently
+      const [creditResponse, debitResponse, onchainResponse] = await Promise.allSettled([
+        creditBalance({}),
+        debitBalance({}),
+        onchainBalance({}),
+      ])
+
+      const newBalances: Record<string, BalanceDisplay> = {
+        bitcoin: { amount: "0", unit: "BTC" },
+        eiou: { amount: "0", unit: "eIOU" },
+        credit: { amount: "0", unit: "credit" },
+        debit: { amount: "0", unit: "debit" },
+      }
+
+      // Handle credit balance response
+      if (creditResponse.status === "fulfilled" && !isErrorResponse(creditResponse.value)) {
+        const creditData = creditResponse.value.data
+        if (creditData && "amount" in creditData && "unit" in creditData) {
+          newBalances.credit = {
+            amount: String(creditData.amount),
+            unit: String(creditData.unit),
+          }
+        }
+      } else {
+        console.warn(
+          "Failed to fetch credit balance:",
+          creditResponse.status === "rejected" ? creditResponse.reason : creditResponse.value.error,
+        )
+      }
+
+      // Handle debit balance response
+      if (debitResponse.status === "fulfilled" && !isErrorResponse(debitResponse.value)) {
+        const debitData = debitResponse.value.data
+        if (debitData && "amount" in debitData && "unit" in debitData) {
+          newBalances.debit = {
+            amount: String(debitData.amount),
+            unit: String(debitData.unit),
+          }
+        }
+      } else {
+        console.warn(
+          "Failed to fetch debit balance:",
+          debitResponse.status === "rejected" ? debitResponse.reason : debitResponse.value.error,
+        )
+      }
+
+      if (onchainResponse.status === "fulfilled" && !isErrorResponse(onchainResponse.value)) {
+        const onchainData = onchainResponse.value.data
+        if (onchainData && typeof onchainData === "object" && "confirmed" in onchainData) {
+          console.log("Onchain balance:", onchainData.confirmed)
+          newBalances.bitcoin = {
+            amount: String(onchainData.confirmed),
+            unit: "BTC",
+          }
+        }
+      } else {
+        console.warn(
+          "Failed to fetch onchain balance:",
+          onchainResponse.status === "rejected" ? onchainResponse.reason : onchainResponse.value.error,
+        )
+      }
+
+      setBalances(newBalances)
+    } catch (error) {
+      const errorMessage = error instanceof Error ? error.message : "Unknown error occurred"
+      setError(errorMessage)
+      console.error("Failed to fetch balances:", error)
+    } finally {
+      setIsLoading(false)
+    }
+  }, [])
+
+  useEffect(() => {
+    void fetchAllBalances()
+
+    const interval = setInterval(() => void fetchAllBalances(), 30000)
```

Comment:
All good here, just a suggestion to separate the decimals here for better readability xD

```suggestion
    const interval = setInterval(() => void fetchAllBalances(), 30_000)
```